### ISSUE TITLE
[Quest] Congratulations - bonus quest missing

### ISSUE THREAD

**Issue by truePaladin** [WoWManiaUK/Redemption#2707]
Having completed the daily quest "Troll patrol" in Zul'Drak 5 times within the time limit (5-7 minutes to spare), the following bonus quest "Congratulations" was not made available to me.

*npc: https://www.wowhead.com/npc=28039
*quest 1: https://www.wowhead.com/quest=12587/troll-patrol
*quest 2: https://www.wowhead.com/quest=12604/congratulations

**Comment by wow-mania** [comment]
Please check:
http://www.wow-mania.com/armory/?quest=12604
Read the requirements.

**Comment by truePaladin** [comment]
I met the requirements, with 5-7 minutes (of the buff) to spare.

**Comment by wow-mania** [comment]
I will re-check on dev quick and let you know.

**Comment by truePaladin** [comment]
So, in case it would help, I got into the habit of taking screenshots just before turn-in. Here is my latest screenshot before not receiving the bonus quest:

![wowscrnshot_090918_225352](https://user-images.githubusercontent.com/43026083/45278875-d6aa4a80-b483-11e8-8066-4dfce91d8144.jpg)

After turn-in, the "on patrol" buff just vanished, and no further quests were available.

**Comment by hammereddann** [comment]
![image](https://user-images.githubusercontent.com/122736582/221187030-09dfc992-bfa5-4ac6-8287-66cb70a1b254.png)
I did this quest just now and handed in with 6 minutes to spare and I didn't get the Congratulations! quest
![image](https://user-images.githubusercontent.com/122736582/221187269-74291546-a2b2-4915-8b6f-c549d72242d9.png)

**Comment by hammereddann** [comment]
![image](https://user-images.githubusercontent.com/122736582/221187395-cb65aaf6-17b1-469e-827c-0fadefc92d7f.png)

**Comment by GMZinius** [comment]
Thank you @hammereddann
I have reopened this to be looked into and confirmed by the test team

**Comment by WhitephoenixTheBunny** [comment]
Test on PTR
![image](https://user-images.githubusercontent.com/31921098/221230971-5699d4d2-c840-4ac3-87c8-4782e252c852.png)
once I turned the quest the NPC Removed my aura

**Comment by WhitephoenixTheBunny** [comment]
Day 2 of testing

![image](https://user-images.githubusercontent.com/31921098/221367259-cded5946-75bc-46c5-b5b1-c100fa0745e3.png)

While turning the [quest ](https://www.wow-mania.com/armory/?quest=12501 )the buff [On Patrol](https://www.wow-mania.com/armory/?spell=51573) is removed, like that the quest [Congratulations!](https://www.wow-mania.com/armory/?quest=12604) is not offered

**Comment by Rushor** [comment]
affected script line:

```sql
(@ENTRY,0,10,0,20,0,100,0,12587,0,0,0,0,28,51573,0,0,0,0,0,7,0,0,0,0,0,0,0,"Commander Kunz - On Quest Reward Congratulations - Remove Aura On Patrol"),
(@ENTRY,0,11,0,20,0,100,0,12563,0,0,0,0,28,53707,0,0,0,0,0,7,0,0,0,0,0,0,0,"Commander Kunz - On Quest Reward Troll Patrol - Remove Aura On Patrol Heartbeat Script"),
(@ENTRY,0,12,0,20,0,100,0,12501,0,0,0,0,28,53707,0,0,0,0,0,7,0,0,0,0,0,0,0,"Commander Kunz - On Quest Reward Troll Patrol - Remove Aura On Patrol Heartbeat Script"),
(@ENTRY,0,13,0,20,0,100,0,12604,0,0,0,0,28,53707,0,0,0,0,0,7,0,0,0,0,0,0,0,"Commander Kunz - On Quest Reward Troll Patrol - Remove Aura On Patrol Heartbeat Script");
```

**Comment by Rushor** [comment]
for me it worked, once troll patrol is completed i can reward the bonus quest and the troll patrol quest

https://i.imgur.com/lQvrmFF.jpeg
https://i.imgur.com/tTDPcsI.jpeg

Can you retest this with those quests aswell: 12587 12501 12563

There are multiple troll patrol quests

**Comment by Echinaceax** [comment]
@Rushor extra info from Phoenix is in dev

**Comment by fvicaria** [comment]
Any updates on this one?

**Comment by Echinaceax** [comment]
> Any updates on this one?

Yes in dev channel @fvicaria

**Comment by WhitephoenixTheBunny** [comment]
Still waiting on feedback to test a possible fix.
The quest do not show up when we turn the quest, as the NPC remove the buff

**Comment by Rushor** [comment]
On hold for now - cannot find the reason atm. weill reopen when i found it
